contracts/EpochTokenLocker.sol

```diff
@@ -205,8 +205,12 @@ contract EpochTokenLocker {
         balanceStates[user][token].balance = balanceStates[user][token].balance.add(amount);
     }
 
-    function subtractBalance(address user, address token, uint256 amount) internal {
+    function subtractBalanceWithPendingWithdrawCheck(address user, address token, uint256 amount) internal {
```

Comment:
I'm wondering what the most clear naming is here. I think the default should be subtracting with a check, so maybe the other method could be called subtractBalanceUnchecked and have a doc string describing the potential attack. This one would then stay "subtractBalance"

Alternatively we could just do the balance check right in BatchExchange (but I kind of like that we warn potential other users of EpochTokenLocker) about it.

Reply:
Nice! I like.